File: hackerrank/30-days-of-code/class-vs-instance/class-vs-instance.cpp
#include <iostream>
#include <bits/stdc++.h>

using namespace std;

class Person{
    public:
        int age;
        Person(int initialAge);
        void amIOld();
        void yearPasses();
    };

    Person::Person(int initialAge){
        age = initialAge;
        if (age < 0) {
            cout << "Age is not valid, setting age to 0." << endl;
            age = 0;
        }
        // Add some more code to run some checks on initialAge
    }

    void Person::amIOld(){
        if (age < 13) {
            cout << "You are young." << endl;
        } else if (age >= 13 && age < 18) {
            cout << "You are a teenager." << endl;
        } else {
            cout << "You are old." << endl;
        }
        
        // Do some computations in here and print out the correct statement to the console 
    }

    void Person::yearPasses(){
        age++;
        // Increment the age of the person in here

    }

int main(){
    int t;
	int age;
    cin >> t;
    for(int i=0; i < t; i++) {
    	cin >> age;
        Person p(age);
        p.amIOld();
        for(int j=0; j < 3; j++) {
        	p.yearPasses(); 
        }
        p.amIOld();
      
		cout << '\n';
    }

    return 0;
}

File: hackerrank/30-days-of-code/class-vs-instance/ClassVsInstance.java
import java.io.*;
import java.util.*;

public class ClassVsInstance {
    private int age;
  
	public ClassVsInstance(int initialAge) {
        this.age = initialAge;
        if (this.age < 0){
            System.out.println("Age is not valid, setting age to 0.");
            this.age = 0;
        }
  		// Add some more code to run some checks on initialAge
	}

	public void amIOld() {
  		// Write code determining if this person's age is old and print the correct statement:
        if (this.age < 13) {
            System.out.println("You are young.");
        } else if ((this.age >= 13) && (this.age < 18)){
            System.out.println("You are a teenager.");
        } else {
            System.out.println("You are old.");
        }   
	}

	public void yearPasses() {
        this.age++;
  		// Increment this person's age.
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int T = sc.nextInt();
		for (int i = 0; i < T; i++) {
			int age = sc.nextInt();
			ClassVsInstance p = new ClassVsInstance(age);
			p.amIOld();
			for (int j = 0; j < 3; j++) {
				p.yearPasses();
			}
			p.amIOld();
			System.out.println();
        }
		sc.close();
    }
}
